File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_py/code_output/WeatherSystem.py
class WeatherSystem:
    def __init__(self, city: str):
        self.temperature: float = None
        self.weather: str = None
        self.city: str = city
        self.weatherList: dict = {}

    def setCity(self, city: str) -> None:
        self.city = city

    def getCity(self) -> str:
        return self.city

    def setTemperature(self, temperature: float) -> None:
        self.temperature = temperature

    def celsiusToFahrenheit(self) -> float:
        return (self.temperature * 9 / 5) + 32

    def fahrenheitToCelsius(self) -> float:
        return (self.temperature - 32) * 5 / 9

    def query(self, weatherList: dict, tmpUnits: str) -> list:
        self.weatherList = weatherList
        if self.city not in weatherList:
            return [False]

        cityWeather = weatherList[self.city]
        self.temperature = cityWeather["temperature"]
        self.weather = cityWeather["weather"]
        currentUnits = cityWeather["temperature units"]

        if currentUnits != tmpUnits:
            if tmpUnits == "celsius":
                self.temperature = self.fahrenheitToCelsius()
            elif tmpUnits == "fahrenheit":
                self.temperature = self.celsiusToFahrenheit()

        return [self.temperature, self.weather]

File: data/derived_cot/rq1_traces/translation/glm5.1/java_to_cpp/code_output/WeatherSystem.cpp
#include <string>
#include <unordered_map>
#include <any>
#include <vector>
#include <utility>

namespace org::example {

class WeatherSystem {
private:
    double temperature = 0.0;
    std::string weather;
    std::string city;
    std::unordered_map<std::string, std::unordered_map<std::string, std::any>> weatherList;

public:
    explicit WeatherSystem(std::string city) : city(std::move(city)) {}

    void setCity(std::string city) {
        this->city = std::move(city);
    }

    std::string getCity() const {
        return this->city;
    }

    void setTemperature(double temperature) {
        this->temperature = temperature;
    }

    double celsiusToFahrenheit() const {
        return (this->temperature * 9.0 / 5.0) + 32.0;
    }

    double fahrenheitToCelsius() const {
        return (this->temperature - 32.0) * 5.0 / 9.0;
    }

    std::vector<std::any> query(const std::unordered_map<std::string, std::unordered_map<std::string, std::any>>& weatherList, const std::string& tmpUnits) {
        this->weatherList = weatherList; // Copy the map into the member variable
        
        if (this->weatherList.find(this->city) == this->weatherList.end()) {
            return {false}; // Equivalent to new Object[]{false}
        }

        const auto& cityWeather = this->weatherList.at(this->city);
        
        // std::any_cast will throw std::bad_any_cast if the key doesn't exist or type mismatches,
        // which is behaviorally equivalent to Java's NullPointerException / ClassCastException here.
        this->temperature = std::any_cast<double>(cityWeather.at("temperature"));
        this->weather = std::any_cast<std::string>(cityWeather.at("weather"));
        std::string currentUnits = std::any_cast<std::string>(cityWeather.at("temperature units"));

        if (currentUnits != tmpUnits) {
            if (tmpUnits == "celsius") {
                this->temperature = fahrenheitToCelsius();
            } else if (tmpUnits == "fahrenheit") {
                this->temperature = celsiusToFahrenheit();
            }
        }

        return {this->temperature, this->weather}; // Equivalent to new Object[]{this.temperature, this.weather}
    }
};

} // namespace org::example